spec/examples/testscript/testscript_fhirpath_test.json

```diff
@@ -0,0 +1,152 @@
+  "copyright": "© HL7.org 2011+",
+  "metadata": {
+    "link": [
+      {
+        "url": "http://hl7.org/fhir/patient.html",
+        "description": "Demographics and other administrative information about an individual or animal receiving care or other health-related services."
+      }
+    ],
+    "capability": [
+      {
+        "required": true,
+        "validated": false,
+        "description": "Patient Search Operation",
+        "link": [
+          "http://hl7.org/fhir/http.html#search"
+        ],
+        "capabilities": "http://hl7.org/fhir/CapabilityStatement/example"
+      }
+    ]
+  },
+  "fixture": [
+    {
+      "id": "fixture-patient-bundle",
+      "autocreate": false,
+      "autodelete": false,
+      "resource": {
+        "reference": "fixtures/Patient/example_bundle.json"
+      }
+    },
+    {
+      "id": "fixture-patient-bundle-entry",
+      "autocreate": false,
+      "autodelete": false,
+      "resource": {
+        "reference": "fixtures/Patient/example_bundle_entry.json"
+      }
+    }
+  ],
+  "variable": [
+    {
+      "name": "PatientSearchBundleTotal",
+      "description": "Evaluate the returned Patient searchset Bundle.total value",
+      "expression": "Bundle.total.toInteger()"
+    }
+  ],
+  "test": [
+    {
+      "id": "02-PatientSearchDynamic",
+      "name": "Patient Search Dynamic",
+      "description": "Search for Patient resources using the user defined dynamic variables ${PatientSearchFamilyName} and ${PatientSearchGivenName} and validate response.",
+      "action": [
+        {
+          "assert": {
+            "description": "Confirm Bundle.total.toInteger() >= entry.count()",
+            "expression": "Bundle.total.toInteger() >= entry.count()",
+            "value": "true",
+            "operator": "greaterThan",
+            "sourceId": "fixture-patient-bundle",
+            "warningOnly": false
+          }
+        },
+        {
+          "assert": {
+            "description": "Confirm total is 393827 as integer in value",
+            "expression": "Bundle.total",
+            "value": "393827",
+            "sourceId": "fixture-patient-bundle",
+            "warningOnly": false
+          }
+        },
+        {
+          "assert": {
+            "description": "Confirm Bundle.entry[0].resource.children().count() = Patient.children().count()",
+            "compareToSourceExpression": "Patient.children().count()",
+            "expression": "Bundle.entry[0].resource.children().count()",
+            "operator": "equals",
+            "compareToSourceId": "fixture-patient-bundle-entry",
+            "sourceId": "fixture-patient-bundle",
+            "warningOnly": false
+          }
+        },
+        {
+          "assert": {
+            "description": "Confirm Bundle.entry[0].resource.children() = Patient.children()",
+            "compareToSourceExpression": "Patient.children()",
+            "expression": "Bundle.entry[0].resource.children()",
```

Comment:
This is a reasonable test case, but the difference in operation between the internal and external FHIR path engines is not appropriate.  If the external engine is not allowed to return a list, then the internal engine should not be able to do so either. The two should only have a different result if what they actually return is different and if you run these expressions on the internal and external engines, both return the same lists, with the external engine adding the wrapping as we know.  Easiest way to move forward is to add the restriction to the internal engine.